### PR TITLE
Provide new check for memory overallocation in FreeBSD

### PR DESCRIPTION
**Pull request by TuxPowered42**

### CODE REVIEW

File: src/check_freebsd_memory.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    'wired': 'v_wire',
+}
+
+# Speci
+NAGIOS_CODES = {
+    'OK': 0,
+    'WARNING': 1,
+    'CRITICAL': 2,
+    'UNKNOWN': 3,
+}
```
Comment (hasegeli) [inline src/check_freebsd_memory.py]: I would (and do all the time) make this a class:
```python
class ExitCodes:
    OK = 0,
    WARNING = 1,
...
```

File: src/check_freebsd_memory.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+    print(exit_multiline)
+    return exit_code
+
+
+def parse_args():
+    parser = ArgumentParser()
+    for memory_type in MEMORY_TYPES.keys():
+        parser.add_argument(
+            '--{}-warning'.format(memory_type), default='50',
```
Comment (hasegeli) [inline src/check_freebsd_memory.py]: `argparse` do not cast the default values.  It shouldn't be in quotes.

File: src/check_freebsd_memory.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+import sys
+
+
+# Translate human-readable names to sysctl
+MEMORY_TYPES = {
+    'active': 'v_active',
+    'wired': 'v_wire',
+}
+
+# Speci
```
Comment (hasegeli) [inline src/check_freebsd_memory.py]: ?

File: src/check_freebsd_memory.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    print(exit_multiline)
+    return exit_code
+
+
+def parse_args():
+    parser = ArgumentParser()
+    for memory_type in MEMORY_TYPES.keys():
+        parser.add_argument(
+            '--{}-warning'.format(memory_type), default='50',
+            required=True, metavar='%', type=int,
```
Comment (hasegeli) [inline src/check_freebsd_memory.py]: Required is unnecessary with a default, I believe.

File: src/check_freebsd_memory.py (hunk 2 above)
Comment (TuxPowered42) [reply src/check_freebsd_memory.py]: Fixed.

File: src/check_freebsd_memory.py (hunk 4 above)
Comment (TuxPowered42) [reply src/check_freebsd_memory.py]: Fixed.

**Pull request merged**